### PR TITLE
A 4 - class voice based mood detector which achieves an accuracy of 92 % using a simple 2 convolutional layered model .

### LINKED ISSUE #1306: Adding a Voice based mood detector model
Adding a 4 class mood detector model with 89% accuracy
@Niketkumardheeryan could you please assign it to me

### PR DESCRIPTION
**Pull request by yashashwanidixit**
Resolves  #1306 
Introduces a PyTorch speech emotion recognition pipeline classifying audio into Happy, Sad, Angry, and Neutral states. It trains on mixed TESS and RAVDESS data, achieving a verified 91.38% test accuracy on a leak-free split.  Key Architecture & ImplementationsLeak-Free Pipeline: Audio paths are isolated into Train (70%), Val (15%), and Test (15%) splits before processing to guarantee clean generalization.  Feature Extraction: Standardized to 16,000Hz sampling rate using 40 MFCC coefficients padded/sliced to 128 frames.  Dynamic Augmentation: Applies Gaussian white noise injection and SpecAugment (time/frequency masking) exclusively to training samples.  Model Structure (emotionCNN): A 2D CNN using dilated convolutions to expand temporal receptive fields, integrated with BatchNorm2d, Dropout2d, and AdaptiveAvgPool2d.  Optimization: Optimized via Adam with a ReduceLROnPlateau scheduler to scale down the learning rate as validation loss flattens.  PerformanceTest Accuracy: 91.38%  Validation Loss: Smooth convergence down to 0.2740.  
image
image

### CODE REVIEW

**Comment by Sibam-Paul** [comment]
using the model can u test sample voice @yashashwanidixit

**Comment by Sibam-Paul** [comment]
like we have onlly models in the repo need some application @yashashwanidixit

**Comment by yashashwanidixit** [comment]
so should i just edit the notebook to include 4 sample voices(happy,sad,neutral and angry) and then display the results of the model on these signals ? or should i add a streamlit app ?

**Comment by Sibam-Paul** [comment]
yeah streamlit would be good like if the person can upload the voice and it will give results

**Comment by yashashwanidixit** [comment]
@Sibam-Paul I have added the streamlit webapp .Pls inform if there's something else to change

**Comment by Sibam-Paul** [comment]
@yashashwanidixit just saw there are 

https://github.com/Niketkumardheeryan/ML-CaPsule/tree/master/Speech_Emotion_Recognition <- this one is most similar to ur project 

https://github.com/Niketkumardheeryan/ML-CaPsule/tree/master/emotion-based%20music%20recommendtion

**Comment by Sibam-Paul** [comment]
@yashashwanidixit can add a streamlit to the existing one can't have similar project

**Comment by yashashwanidixit** [comment]
image





@Sibam-Paul i was trying to push my code i am getting this error
i have alsso created an issue regarding the same

**Comment by yashashwanidixit** [comment]
@Sibam-Paul the first project that u have mentioned has an accuracy of only under 52 %(whilst my model has a robust accuracy of 91.38 percent)and for the second one the model notebook isnt opening .I could have added a streamit app for the first one but the model isnt good as only one dataset-RAVDESS is used whereas have i used both RAVDESS and TESS .So this is not like an entirely new project but its essential for better ser standard for this repo.

**Comment by yashashwanidixit** [comment]
@Sibam-Paul i have furthur improved the model.Current accuracy  91.30%. Along with this i have also added fdr ,mad and cosine similarity comparison for different lifter values which can be modified for better performance of the model.
I request you to consider this or suggest changes considering that the other projects on the same topic in the repo currently are not satisfactory as presented why in my previous comment.

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]
https://github.com/Niketkumardheeryan/ML-CaPsule/tree/master/Emotion%20Recognition%20Based%20on%20NLP

if u can include inside this folder it will be ok 

@yashashwanidixit

**Comment by yashashwanidixit** [comment]
@Sibam-Paul the folder  you have mentioned already contains a different model should i add my separate model ,readme,app,requirements.txt inside that folder itself ?

File: Voice_Based_Mood_Detector/README.md
Comment (Sibam-Paul) [inline Voice_Based_Mood_Detector/README.md:1]: give me the link of the dataset

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]
@yashashwanidixit

File: Voice_Based_Mood_Detector/README.md
Comment (yashashwanidixit) [reply Voice_Based_Mood_Detector/README.md:1]: @Sibam-Paul very sorry for the late reply - i used tess and ravdess dataset from kaggle itself. i used 96 audio files per emotion per dataset- so total 192 files per emotion. i created a tess and ravdess dataset pool to get random audio files ...it is there in the jupyter notebook .
ravdess-kaggle.com/datasets/uwrfkaggler/ravdess-emotional-speech-audio
tess-kaggle.com/datasets/ejlok1/toronto-emotional-speech-set-tess
Also could u remove the noactivity-twarnign tag

Comment (yashashwanidixit) [reply Voice_Based_Mood_Detector/README.md:1]: @Sibam-Paul could u pls guide what should i do

**Comment by yashashwanidixit** [comment]
@Sibam-Paul I request you to please consider my pull requests and tell me on how can i proceed with the project . Should i upload my existing  project files in the already existing project ?

**Comment by yashashwanidixit** [comment]
Hi @Niketkumardheeryan  / @Sibam-Paul , following up on this PR. The stale bot closed it due to inactivity while waiting for feedback on folder organization. Could you please reopen this PR or advise if I should move the files into Emotion Recognition Based on NLP as discussed? Thank you
